src/sentry/models/notificationsetting.py

```diff
@@ -61,6 +62,8 @@ def provider_str(self) -> str:
     target = FlexibleForeignKey(
         "sentry.Actor", db_index=True, unique=False, null=False, on_delete=models.CASCADE
     )
+    team = FlexibleForeignKey("sentry.Team", null=True, db_index=False)
```

Comment:
I don't think this can be a foreign key as we'll need to break it again soon. Could we use HybridCloudForeignKey here instead.